2025-05-03 18:40:55 - INFO - Checking for leader...
2025-05-03 18:40:55 - INFO - [SETUP] Lobby server started on port 60001
2025-05-03 18:41:00 - INFO - Leader found: 127.0.0.1:50000
2025-05-03 18:41:00 - INFO - Info came from: 127.0.0.1:50000
2025-05-03 18:41:07 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:41:07 - INFO - [MAIN] Received request: action: JOIN_LOBBY
username: "nick"

2025-05-03 18:41:07 - INFO - [MAIN] nick connected.
2025-05-03 18:41:09 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:41:09 - INFO - [MAIN] Received request: action: JOIN_LOBBY
username: "nick2"

2025-05-03 18:41:09 - INFO - [MAIN] nick2 connected.
2025-05-03 18:41:10 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:41:10 - INFO - [MAIN] Received request: action: SEND_VOTE
username: "nick"
vote: true

2025-05-03 18:41:11 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:41:11 - INFO - [MAIN] Received request: action: SEND_VOTE
username: "nick2"
vote: true

2025-05-03 18:41:26 - ERROR - [MAIN] Error handling requests at line 217: Traceback (most recent call last):
  File "/Users/nickydtricky/Desktop/Harvard/Senior Year/Semester 2/CS2620/cs2620_poker/server_lobby.py", line 912, in handle_requests
    for req in request_iterator:
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 543, in __next__
    return self._next()
           ^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 535, in _next
    request = self._look_for_request()
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 517, in _look_for_request
    _raise_rpc_error(self._state)
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 217, in _raise_rpc_error
    raise rpc_error
grpc.RpcError

2025-05-03 18:41:26 - INFO - [MAIN] nick disconnected.
2025-05-03 18:41:29 - ERROR - [MAIN] Error handling requests at line 217: Traceback (most recent call last):
  File "/Users/nickydtricky/Desktop/Harvard/Senior Year/Semester 2/CS2620/cs2620_poker/server_lobby.py", line 912, in handle_requests
    for req in request_iterator:
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 543, in __next__
    return self._next()
           ^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 535, in _next
    request = self._look_for_request()
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 517, in _look_for_request
    _raise_rpc_error(self._state)
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 217, in _raise_rpc_error
    raise rpc_error
grpc.RpcError

2025-05-03 18:41:29 - INFO - [MAIN] nick2 disconnected.
2025-05-03 18:41:32 - ERROR - Error receiving response: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Cancelling all calls"
	debug_error_string = "UNKNOWN:Error received from peer  {created_time:"2025-05-03T18:41:32.217047-04:00", grpc_status:14, grpc_message:"Cancelling all calls"}"
>
2025-05-03 18:41:32 - INFO - Checking for leader...
2025-05-03 18:43:24 - INFO - Checking for leader...
2025-05-03 18:43:24 - INFO - [SETUP] Lobby server started on port 60001
2025-05-03 18:43:29 - INFO - Leader found: 127.0.0.1:50001
2025-05-03 18:43:29 - INFO - Info came from: 127.0.0.1:50000
2025-05-03 18:43:43 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:43:43 - INFO - [MAIN] Received request: action: JOIN_LOBBY
username: "nick"

2025-05-03 18:43:43 - INFO - [MAIN] nick connected.
2025-05-03 18:43:49 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:43:49 - INFO - [MAIN] Received request: action: JOIN_LOBBY
username: "nick2"

2025-05-03 18:43:49 - INFO - [MAIN] nick2 connected.
2025-05-03 18:43:50 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:43:50 - INFO - [MAIN] Received request: action: SEND_VOTE
username: "nick"
vote: true

2025-05-03 18:43:51 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:43:51 - INFO - [MAIN] Received request: action: SEND_VOTE
username: "nick2"
vote: true

2025-05-03 18:43:53 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:43:53 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:43:54 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:43:54 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:43:55 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:43:55 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:43:57 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:43:57 - INFO - [MAIN] Received request: action: PLAY_MOVE
player_action: EXCHANGE
card_exchange_idx: 0
card_exchange_idx: 1
card_exchange_idx: 1
card_exchange_idx: 0
card_exchange_idx: 0

2025-05-03 18:44:11 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:44:11 - INFO - [MAIN] Received request: action: PLAY_MOVE
player_action: EXCHANGE
card_exchange_idx: 1
card_exchange_idx: 0
card_exchange_idx: 1
card_exchange_idx: 0
card_exchange_idx: 0

2025-05-03 18:44:15 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:44:15 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:44:16 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:44:16 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:44:17 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:44:17 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:44:18 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:44:18 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:44:19 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:44:19 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:44:20 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:44:20 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:44:21 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:44:21 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:44:22 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:44:22 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:44:22 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:44:22 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:44:23 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:44:23 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:44:24 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:44:24 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:44:25 - ERROR - [MAIN] Error handling requests at line 217: Traceback (most recent call last):
  File "/Users/nickydtricky/Desktop/Harvard/Senior Year/Semester 2/CS2620/cs2620_poker/server_lobby.py", line 912, in handle_requests
    for req in request_iterator:
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 543, in __next__
    return self._next()
           ^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 535, in _next
    request = self._look_for_request()
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 517, in _look_for_request
    _raise_rpc_error(self._state)
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 217, in _raise_rpc_error
    raise rpc_error
grpc.RpcError

2025-05-03 18:44:25 - INFO - [MAIN] nick disconnected.
2025-05-03 18:44:26 - ERROR - [MAIN] Error handling requests at line 217: Traceback (most recent call last):
  File "/Users/nickydtricky/Desktop/Harvard/Senior Year/Semester 2/CS2620/cs2620_poker/server_lobby.py", line 912, in handle_requests
    for req in request_iterator:
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 543, in __next__
    return self._next()
           ^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 535, in _next
    request = self._look_for_request()
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 517, in _look_for_request
    _raise_rpc_error(self._state)
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 217, in _raise_rpc_error
    raise rpc_error
grpc.RpcError

2025-05-03 18:44:26 - INFO - [MAIN] nick2 disconnected.
2025-05-03 18:46:45 - ERROR - Error receiving response: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Cancelling all calls"
	debug_error_string = "UNKNOWN:Error received from peer  {grpc_message:"Cancelling all calls", grpc_status:14, created_time:"2025-05-03T18:46:45.122247-04:00"}"
>
2025-05-03 18:46:45 - INFO - Checking for leader...
2025-05-03 18:47:06 - INFO - Checking for leader...
2025-05-03 18:47:06 - INFO - [SETUP] Lobby server started on port 60001
2025-05-03 18:47:11 - INFO - Leader found: 127.0.0.1:50000
2025-05-03 18:47:11 - INFO - Info came from: 127.0.0.1:50000
2025-05-03 18:47:28 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:28 - INFO - [MAIN] Received request: action: JOIN_LOBBY
username: "nick"

2025-05-03 18:47:28 - INFO - [MAIN] nick connected.
2025-05-03 18:47:33 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:33 - INFO - [MAIN] Received request: action: JOIN_LOBBY
username: "nick2"

2025-05-03 18:47:33 - INFO - [MAIN] nick2 connected.
2025-05-03 18:47:34 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:34 - INFO - [MAIN] Received request: action: SEND_VOTE
username: "nick2"
vote: true

2025-05-03 18:47:34 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:34 - INFO - [MAIN] Received request: action: SEND_VOTE
username: "nick"
vote: true

2025-05-03 18:47:35 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:35 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:47:36 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:36 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:47:36 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:36 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:47:40 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:40 - INFO - [MAIN] Received request: action: PLAY_MOVE
player_action: EXCHANGE
card_exchange_idx: 1
card_exchange_idx: 0
card_exchange_idx: 0
card_exchange_idx: 1
card_exchange_idx: 0

2025-05-03 18:47:43 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:43 - INFO - [MAIN] Received request: action: PLAY_MOVE
player_action: EXCHANGE
card_exchange_idx: 0
card_exchange_idx: 0
card_exchange_idx: 0
card_exchange_idx: 1
card_exchange_idx: 1

2025-05-03 18:47:44 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:44 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:47:45 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:45 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:47:46 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:46 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:47:47 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:47 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:47:56 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:56 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:47:58 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:58 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:47:59 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:47:59 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:48:00 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:48:00 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:48:18 - ERROR - [MAIN] Error handling requests at line 217: Traceback (most recent call last):
  File "/Users/nickydtricky/Desktop/Harvard/Senior Year/Semester 2/CS2620/cs2620_poker/server_lobby.py", line 907, in handle_requests
    for req in request_iterator:
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 543, in __next__
    return self._next()
           ^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 535, in _next
    request = self._look_for_request()
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 517, in _look_for_request
    _raise_rpc_error(self._state)
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 217, in _raise_rpc_error
    raise rpc_error
grpc.RpcError

2025-05-03 18:48:18 - INFO - [MAIN] nick disconnected.
2025-05-03 18:48:20 - ERROR - [MAIN] Error handling requests at line 217: Traceback (most recent call last):
  File "/Users/nickydtricky/Desktop/Harvard/Senior Year/Semester 2/CS2620/cs2620_poker/server_lobby.py", line 907, in handle_requests
    for req in request_iterator:
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 543, in __next__
    return self._next()
           ^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 535, in _next
    request = self._look_for_request()
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 517, in _look_for_request
    _raise_rpc_error(self._state)
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 217, in _raise_rpc_error
    raise rpc_error
grpc.RpcError

2025-05-03 18:48:20 - INFO - [MAIN] nick2 disconnected.
2025-05-03 18:48:23 - ERROR - Error receiving response: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Cancelling all calls"
	debug_error_string = "UNKNOWN:Error received from peer  {created_time:"2025-05-03T18:48:23.01782-04:00", grpc_status:14, grpc_message:"Cancelling all calls"}"
>
2025-05-03 18:48:23 - INFO - Checking for leader...
2025-05-03 18:48:32 - INFO - Checking for leader...
2025-05-03 18:48:32 - INFO - [SETUP] Lobby server started on port 60001
2025-05-03 18:48:37 - INFO - Leader found: 127.0.0.1:50000
2025-05-03 18:48:37 - INFO - Info came from: 127.0.0.1:50000
2025-05-03 18:48:50 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:48:50 - INFO - [MAIN] Received request: action: JOIN_LOBBY
username: "nick"

2025-05-03 18:48:50 - INFO - [MAIN] nick connected.
2025-05-03 18:48:56 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:48:56 - INFO - [MAIN] Received request: action: JOIN_LOBBY
username: "nick2"

2025-05-03 18:48:56 - INFO - [MAIN] nick2 connected.
2025-05-03 18:48:57 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:48:57 - INFO - [MAIN] Received request: action: SEND_VOTE
username: "nick"
vote: true

2025-05-03 18:48:58 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:48:58 - INFO - [MAIN] Received request: action: SEND_VOTE
username: "nick2"
vote: true

2025-05-03 18:48:59 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:48:59 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:49:00 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:49:00 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:49:00 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:49:00 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:49:02 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:49:02 - INFO - [MAIN] Received request: action: PLAY_MOVE
player_action: EXCHANGE
card_exchange_idx: 0
card_exchange_idx: 1
card_exchange_idx: 1
card_exchange_idx: 0
card_exchange_idx: 0

2025-05-03 18:49:03 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:49:03 - INFO - [MAIN] Received request: action: PLAY_MOVE
player_action: EXCHANGE
card_exchange_idx: 0
card_exchange_idx: 0
card_exchange_idx: 0
card_exchange_idx: 1
card_exchange_idx: 1

2025-05-03 18:49:04 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:49:04 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:49:05 - INFO - [MAIN] Size of request: 80 bytes
2025-05-03 18:49:05 - INFO - [MAIN] Received request: action: PLAY_MOVE

2025-05-03 18:49:06 - ERROR - [MAIN] Error handling requests at line 217: Traceback (most recent call last):
  File "/Users/nickydtricky/Desktop/Harvard/Senior Year/Semester 2/CS2620/cs2620_poker/server_lobby.py", line 907, in handle_requests
    for req in request_iterator:
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 543, in __next__
    return self._next()
           ^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 535, in _next
    request = self._look_for_request()
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 517, in _look_for_request
    _raise_rpc_error(self._state)
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 217, in _raise_rpc_error
    raise rpc_error
grpc.RpcError

2025-05-03 18:49:07 - ERROR - [MAIN] Error handling requests at line 217: Traceback (most recent call last):
  File "/Users/nickydtricky/Desktop/Harvard/Senior Year/Semester 2/CS2620/cs2620_poker/server_lobby.py", line 907, in handle_requests
    for req in request_iterator:
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 543, in __next__
    return self._next()
           ^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 535, in _next
    request = self._look_for_request()
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 517, in _look_for_request
    _raise_rpc_error(self._state)
  File "/Users/nickydtricky/miniforge3/envs/manim-use/lib/python3.11/site-packages/grpc/_server.py", line 217, in _raise_rpc_error
    raise rpc_error
grpc.RpcError

2025-05-03 18:51:03 - INFO - Checking for leader...
2025-05-03 18:51:03 - INFO - [SETUP] Lobby server started on port 60001
2025-05-03 18:51:08 - INFO - Leader found: 127.0.0.1:50001
2025-05-03 18:51:08 - INFO - Info came from: 127.0.0.1:50000
2025-05-03 19:12:41 - INFO - Checking for leader...
2025-05-03 19:12:41 - INFO - [SETUP] Lobby server started on port 60001
2025-05-03 19:12:46 - INFO - Leader found: 127.0.0.1:50004
2025-05-03 19:12:46 - INFO - Info came from: 127.0.0.1:50000
2025-05-03 19:14:43 - ERROR - Error receiving response: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Cancelling all calls"
	debug_error_string = "UNKNOWN:Error received from peer  {created_time:"2025-05-03T19:14:43.903192-04:00", grpc_status:14, grpc_message:"Cancelling all calls"}"
>
2025-05-03 19:14:43 - INFO - Checking for leader...
